packages/editor/src/components/Editor.js

```diff
@@ -135,27 +102,50 @@ class BlockEditor extends Component {
   }
 
   render() {
+    const { plugins, theme, components } = this.props
+    const allComponents = {
+      ...defaultBlocks,
+      ...components
+    }
+    const context = {
+      components: allComponents
+    }
+
     return (
       <div style={{ minHeight: '100vh' }}>
-        <Editor
-          {...this.props}
-          ref={editor => (this.editor = editor)}
-          schema={schema}
-          placeholder="Write some MDX..."
-          plugins={plugins}
-          value={this.state.value}
-          onChange={this.handleChange}
-          onKeyDown={this.handleKeyDown}
-          onPaste={this.handlePaste}
-          renderEditor={(_props, _editor, next) => {
-            const children = next()
-
-            return <>{children}</>
-          }}
-        />
+        <Context.Provider value={context}>
+          <Editor
+            {...this.props}
+            ref={editor => (this.editor = editor)}
+            components={allComponents}
+            theme={theme}
+            schema={schema}
+            placeholder="Write some MDX..."
+            plugins={plugins}
+            value={this.state.value}
+            onChange={this.handleChange}
```

Comment:
I think we can avoid duplicative state here and just pass props through if anyone wants to use `onChange`, right?